src/main/java/org/spongepowered/api/text/channel/AbstractMutableMessageChannel.java

```diff
@@ -24,18 +24,19 @@
  */
 package org.spongepowered.api.text.channel;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+
 import java.util.Collection;
 import java.util.Collections;
 import java.util.Set;
 import java.util.WeakHashMap;
 
 /**
- * An abstract implementation of {@link MutableMessageChannel}, using a
- * {@link Set} as the underlying member list
+ * An abstract implementation of {@link MutableMessageChannel}.
  */
 public abstract class AbstractMutableMessageChannel implements MutableMessageChannel {
 
-    protected final Set<MessageReceiver> members;
+    protected final Collection<MessageReceiver> members;
```

Comment:
It's up to the developer to avoid this - by default a `WeakHashMap`-based `Set` is used.